griptape/tools/griptape_cloud_knowledge_base_client/tool.py

```diff
@@ -40,6 +40,24 @@ def query(self, params: dict) -> TextArtifact | ErrorArtifact:
         except exceptions.RequestException as err:
             return ErrorArtifact(str(err))
 
+    def raw_query(self, params: dict) -> ListArtifact | ErrorArtifact:
+        from requests import post, exceptions
+
+        query = params["values"]["query"]
+        url = urljoin(self.base_url.strip("/"), f"/api/knowledge-bases/{self.knowledge_base_id}/query")
+
+        try:
+            response = post(url, json={"query": query, "raw": True}, headers=self.headers)
+            response_body = response.json()
+
+            artifacts: list[BaseArtifact] = []
+            for query_result in response_body.get("result", []):
+                artifacts.append(BaseArtifact.from_json(query_result.meta["artifact"]))
+
+            return ListArtifact(artifacts)
+        except exceptions.RequestException as err:
+            return ErrorArtifact(str(err))
+
```

Comment:
This should be part of the schema for `query`, not a new Activity.

Reply:
Will remain a draft until internal KnowledgeBase API conversation concludes. But updated to be part of `query` for now